### ISSUE TITLE
ICE on nightly when compiling rust-dispatch

### ISSUE THREAD

**Issue by ryanisaacg** [rust-lang/rust#71867]
### Code

```Rust
#[repr(C)]
pub struct dispatch_object_s { _private: [u8; 0] }

extern {
    static _dispatch_queue_attr_concurrent: dispatch_object_s;
}

pub static DISPATCH_QUEUE_CONCURRENT: &'static dispatch_object_s = unsafe { &_dispatch_queue_attr_concurrent };
```

### Meta

`rustc --version --verbose`:
```
rustc 1.42.0 (b8cedc004 2020-03-09)
binary: rustc
commit-hash: b8cedc00407a4c56a3bda1ed605c6fc166655447
commit-date: 2020-03-09
host: x86_64-apple-darwin
release: 1.42.0
LLVM version: 9.0
```

### Error output

```
error: internal compiler error: src/librustc_mir/interpret/validity.rs:396: Unexpected error during ptr inbounds test: cannot read from foreign (extern) static DefId(0:7 ~ playground[ae9d]::[0]::_dispatch_queue_attr_concurrent[0])

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:907:9
```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1069
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1532
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: rustc_driver::report_ice
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:481
  12: std::panicking::begin_panic
  13: rustc_errors::HandlerInner::bug
  14: rustc_errors::Handler::bug
  15: rustc_middle::util::bug::opt_span_bug_fmt::{{closure}}
  16: rustc_middle::ty::context::tls::with_opt::{{closure}}
  17: rustc_middle::ty::context::tls::with_opt
  18: rustc_middle::util::bug::opt_span_bug_fmt
  19: rustc_middle::util::bug::bug_fmt
  20: rustc_mir::interpret::validity::ValidityVisitor<M>::check_safe_pointer
  21: rustc_mir::interpret::validity::ValidityVisitor<M>::try_visit_primitive
  22: rustc_mir::interpret::validity::<impl rustc_mir::interpret::eval_context::InterpCx<M>>::validate_operand_internal
  23: rustc_mir::const_eval::eval_queries::const_eval_validated_provider
  24: rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::const_eval_validated>::compute
  25: rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl
  26: rustc_query_system::query::plumbing::get_query_impl
  27: rustc_mir::const_eval::eval_queries::const_eval_validated_provider
  28: rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::const_eval_validated>::compute
  29: rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl
  30: rustc_query_system::query::plumbing::get_query_impl
  31: rustc_middle::mir::interpret::queries::<impl rustc_middle::ty::context::TyCtxt>::const_eval_global_id
  32: rustc_middle::mir::interpret::queries::<impl rustc_middle::ty::context::TyCtxt>::const_eval_poly
  33: <rustc_lint::BuiltinCombinedLateLintPass as rustc_lint::passes::LateLintPass>::check_item
  34: rustc_hir::intravisit::Visitor::visit_nested_item
  35: rustc_hir::intravisit::walk_crate
  36: rustc_lint::late::late_lint_crate
  37: rustc_data_structures::sync::join
  38: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  39: rustc_session::utils::<impl rustc_session::session::Session>::time
  40: rustc_interface::passes::analysis
  41: rustc_middle::ty::query::<impl rustc_query_system::query::config::QueryAccessors<rustc_middle::ty::context::TyCtxt> for rustc_middle::ty::query::queries::analysis>::compute
  42: rustc_query_system::dep_graph::graph::DepGraph<K>::with_task_impl
  43: rustc_query_system::query::plumbing::get_query_impl
  44: rustc_middle::ty::context::tls::enter_global
  45: rustc_interface::interface::run_compiler_in_existing_thread_pool
  46: scoped_tls::ScopedKey<T>::set
  47: rustc_ast::attr::with_globals
```

</p>
</details>

**Comment by LeSeulArtichaut** [comment]
Ran `cargo-bisect-rustc`:

searched nightlies: from nightly-2020-04-25 to nightly-2020-04-27
regressed nightly: nightly-2020-04-27
searched commits: from https://github.com/rust-lang/rust/commit/0862458dad90a0d80827e22e3f86e33add6d847c to https://github.com/rust-lang/rust/commit/e83f7563495dbe2629b0cbc738afb0808c4482e1
regressed commit: https://github.com/rust-lang/rust/commit/b592b37fa31c0d81ca23fea9fe7998c3f0f4bd0b (#71566)

I'd say the culprit PR is #71140 (cc @oli-obk) but I don't know if it could also be #71392.

**Comment by LeSeulArtichaut** [comment]
This seems a lot like a duplicate of #71612 though

**Comment by jonas-schievink** [comment]
Closing as duplicate.
